File: tests/test_tensor.py
import numpy as np
import pyarrow as pa
import test_tensor_ext as t


def test_tensor():
    arr = np.array([[2, 2, 4], [4, 5, 100]], np.int32)
    tensor = pa.Tensor.from_numpy(arr, dim_names=["dim1", "dim2"])
    assert t.test_tensor(tensor).equals(tensor)

File: tests/test_tensor.cpp
#include "helpers.hpp"

#include <nanobind/nanobind.h>

#include <nanobind_pyarrow/pyarrow_import.h>
#include <nanobind_pyarrow/tensor.h>

namespace nb = nanobind;


NB_MODULE(test_tensor_ext, m)
{
    using namespace nb::literals;
    using namespace nanobind_pyarrow::Testing;

    static nb::detail::pyarrow::ImportPyarrow module;

    m.def("test_tensor", func<arrow::Tensor>());
    m.def("test_int32_tensor", func<arrow::NumericTensor<arrow::Int32Type>>());
}